### PR TITLE
Adding Text pattern to DataGridViewTextBoxCell

### LINKED ISSUE #2786: DataGridView's edit controls don't support UIA Text pattern as required by UIA rules
* .NET Core Version: 
All

* Have you experienced this same bug with .NET Framework?: 
yes

**Problem description:**
DataGridView needs to support the TextPattern. Accessibility Insights reports failures for this issue

**Expected behavior:**
Accessibility Insights should not report failures.

**Minimal repro:**
Create a DataGridView with multiple data cells

### LINKED ISSUE #3658: [Accessibility] Narrator does not read the image description of the existed image in the DataGridViewImageColumn
* .NET Core Version: 
.Net SDK 5.0.100-rc.1.20378.13

* Have you experienced this same bug with .NET Framework?: 
Yes

More Info:
There is a similar bugs: [1159791 ](https://dev.azure.com/devdiv/DevDiv/_workitems/edit/1159791/?triage=true)in VSO.

**Minimal repro:**

1. Extract “[DataGridViewImage.zip](https://github.com/dotnet/winforms/files/4993008/DataGridViewImage.zip)” and open it in VS.
2. Build and run the project.
3. Open Narrator tool, navigate to the first row, Column2, check whether the narrator reads the image description of the image present.

**Expected behavior:**
Narrator should read the image description of the existed image in the DataGridViewImageColumn.

**Actual behavior:**
Narrator does not read the image description of the existed image in the DataGridViewImageColumn.
image

### PR DESCRIPTION
**Pull request by vladimir-krestov**
Fixes #2786
Original Bug: [1159872](https://dev.azure.com/devdiv/DevDiv/_workitems/edit/1159872)
Based on PR #2701

## Proposed changes

- Enable **Text pattern** support for `DataGridViewTextBoxEditingControlAccessibleObject`

## Customer Impact
- Visually impaired users will be able to read and interact with the text content of a DataGridViewTextBoxCell

## Regression? 

- No

## Risk

- Mininmal

### Before
- Narrator does not announce the text navigation and selection of a DataGridViewTextBoxCell

### After
- Narrator announces the text content of a DataGridViewTextBoxCell (navigation, selection)

## Test methodology 

- Manual testing
- CTI
- Unit tests

## Accessibility testing  
- Using Narrator and Inspect

 

## Test environment(s) 

- .Net 5.0 Version: 5.0.0-rc.1.20431.5
- Microsoft Windows [Version 10.0.19041.450]

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/winforms/pull/3866)

### CODE REVIEW

**Review by RussKie — CHANGES_REQUESTED** [review]
Please rebase

File: src/System.Windows.Forms/src/System/Windows/Forms/DataGridViewTextBoxEditingControl.DataGridViewTextBoxEditingControlAccessibleObject.cs [hunk 1]
```diff
@@ -17,9 +17,14 @@ internal class DataGridViewTextBoxEditingControlAccessibleObject : Control.Contr
             ///  The parent is changed when the editing control is attached to another editing cell.
             /// </summary>
             private AccessibleObject? _parentAccessibleObject;
+            private readonly TextBoxBase _owningDataGridViewTextBoxEditingControl;
+            private readonly TextBoxBaseUiaTextProvider _textProvider;
 
             public DataGridViewTextBoxEditingControlAccessibleObject(DataGridViewTextBoxEditingControl ownerControl) : base(ownerControl)
             {
+                _owningDataGridViewTextBoxEditingControl = ownerControl;
```
Comment (RussKie) [inline src/System.Windows.Forms/src/System/Windows/Forms/DataGridViewTextBoxEditingControl.DataGridViewTextBoxEditingControlAccessibleObject.cs:25]: Test for `ownerControl is null`?

**Review by RussKie — CHANGES_REQUESTED** [review]
LGTM module the feedback.

**Comment by vladimir-krestov** [comment]
Testers approved the fix ✔️
There are 2 known issues that will be fixed as a separate fix for PropertyGrid edit and ToolStripTextBox too:
- #2794: Narrator doesn't announce a text mouse selection (a keyboard selection works well)
- #3897: Narrator doesn't show the blue rectangle around an added character when typing

Also, this fix doesn't implement Text pattern for a DataGridViewTextBoxCell in non-edit mode (just selected). Text pattern works only when the cell in **edit** mode.

But the main aim was achieved.

**Review by RussKie — APPROVED** [review]

**Pull request merged**